optuna/_transform.py

```diff
@@ -90,6 +90,14 @@ def column_to_encoded_columns(self) -> List[numpy.ndarray]:
     def encoded_column_to_column(self) -> numpy.ndarray:
         return self._encoded_column_to_column
 
+    @property
+    def num_params(self) -> int:
+        return len(self._column_to_encoded_columns)
+
+    @property
+    def num_features(self) -> int:
+        return len(self.encoded_column_to_column)
```

Comment:
Seems like dead code. Are these lines left intentionally?

Reply:
This property is added as the symmetric counterpart of `num_params`. Even though it isn't currently used in any implementations, it is likely that some `ImportanceEvaluator` will need it in the future.

Reply:
Will it be used for `shap.py` perhaps? If it's not used by any of our evaluators, I'd say we should wait with this introduction until it become necessary (while I understand the symmetry). It's easier to add later. We could leave a TODO comment for future maintainers, if that'd be the case.

Reply:
I don't think `shap.py` needs `num_features`.
I made it a TODO.

Reply:
Sorry for the nitpicking, but that looks good. Thanks. 🙇